src/Definition.php

```diff
@@ -95,4 +96,42 @@ class Definition
      * @var string
      */
     public $documentation;
+
+    /**
+     * Gets the definitons of all ancestor classes
+     *
+     * @return Definition[]
+     */
+    public function getAncestorDefinitions(ReadableIndex $index, bool $includeSelf = false)
+    {
+        if ($includeSelf) {
+            yield $this;
+        }
+        if (is_array($this->extends)) {
+            // iterating once, storing the references and iterating again
+            // guarantees that closest definitions are returned first
+            $definitions = [];
+            foreach ($this->extends as $fqn) {
+                $def = $index->getDefinition($fqn);
+                if ($def !== null) {
+                    yield $def;
```

Comment:
I think it would make sense to yield the FQN as a key here so a second function is not needed